ui/raidboss/data/07-dt/ultimate/futures_rewritten.ts

```diff
@@ -249,7 +278,120 @@ const triggerSet: TriggerSet<Data> = {
+    },
+    {
+      id: 'FRU P2 Quadruple Slap Cleanse',
+      type: 'GainsEffect',
+      netRegex: { effectId: '1042', source: 'Usurper of Frost' },
+      condition: (data) => data.CanCleanse(),
+      infoText: (data, matches, output) =>
+        output.cleanse!({ player: data.party.member(matches.target).nick }),
+      outputStrings: {
+        cleanse: {
+          en: 'Cleanse ${player}',
+        },
+      },
+    },
+    {
+      id: 'FRU P2 Diamond Dust',
+      type: 'StartsUsing',
+      netRegex: { id: '9D05', source: 'Usurper of Frost', capture: false },
+      response: Responses.bigAoe(),
+    },
+    {
+      id: 'FRU P2 Axe/Scythe Kick Collect',
+      type: 'StartsUsing',
+      // 9D0A - Axe Kick (be out), 9D0B - Scythe Kick (be in)
+      netRegex: { id: ['9D0A', '9D0B'], source: 'Oracle\'s Reflection' },
+      // there are 2 actors 180 degrees apart, but we only need to collect one
+      run: (data, matches) => data.p2AxeScytheSafe = matches.id === '9D0A' ? 'out' : 'in',
+    },
+    {
+      id: 'FRU P2 Icicle Impact Initial Collect',
+      type: 'StartsUsing',
+      netRegex: { id: '9D06', source: 'Usurper of Frost' },
+      // there are 2 actors 180 degrees apart, but we only need to collect one
+      condition: (data) => data.p2IcicleImpactStart === 'unknown',
+      suppressSeconds: 1,
+      run: (data, matches) => {
+        const x = parseInt(matches.x);
+        const y = parseInt(matches.y);
+        const dir = Directions.xyTo8DirOutput(x, y, centerX, centerY);
+        data.p2IcicleImpactStart = dir;
+      },
+    },
+    {
+      id: 'FRU P2 House of Light/Frigid Stone',
+      type: 'HeadMarker',
+      netRegex: { id: '0159' },
+      alertText: (data, matches, output) => {
+        data.p2FrigidStoneTargets.push(matches.target);
+        if (data.p2FrigidStoneTargets.length !== 4)
+          return;
+
+        const inOut = data.p2AxeScytheSafe ? output[data.p2AxeScytheSafe]!() : output.unknown!();
 
+        // Assumes that if first Icicle Impacts spawn on cardinals, House of Light baits will also be
+        // cardinals and Frigid Stone puddle drops will be intercards, and vice versa.
+        if (data.p2FrigidStoneTargets.includes(data.me)) {
+          const dir = data.p2IcicleImpactStart === 'unknown'
+            ? output.unknown!()
+            : (isCardinalDir(data.p2IcicleImpactStart)
+              ? output.intercards!()
+              : output.cardinals!());
+          return output.dropPuddle!({ inOut: inOut, dir: dir });
+        }
+
+        const dir = data.p2IcicleImpactStart === 'unknown'
+          ? output.unknown!()
+          : (isCardinalDir(data.p2IcicleImpactStart) ? output.cardinals!() : output.intercards!());
+        return output.baitCleave!({ inOut: inOut, dir: dir });
+      },
+      outputStrings: {
+        dropPuddle: {
+          en: '${inOut} + Far => Drop Puddle (${dir})',
+        },
+        baitCleave: {
+          en: '${inOut} + Close Bait (${dir})',
```

Comment:
I'm not sure why this is `Close Bait` instead of just `Bait`?  There isn't a far bait so there shouldn't be a need to distinguish .

Reply:
Agreed, just having it be `Bait` would be sufficient.

Also, I tested these triggers tonight in-game and the direction here was always `???`, so that probably needs looked at further. Didn't have time tonight to dig into why, but if no one else has looked at it I'll look tomorrow.

Reply:
> Agreed, just having it be `Bait` would be sufficient.
> 
> Also, I tested these triggers tonight in-game and the direction here was always `???`, so that probably needs looked at further. Didn't have time tonight to dig into why, but if no one else has looked at it I'll look tomorrow.

It's likely the issue mentioned above with stale actor names.

Reply:
Yeah, that is almost certainly the cause.  Once I fixed that, I didn't have that issue any longer.  `c29d508` should take care of it.

Reply:
> I'm not sure why this is Close Bait instead of just Bait? There isn't a far bait so there shouldn't be a need to distinguish .

> Agreed, just having it be Bait would be sufficient.

Hmm, I think for me, "Close" is actually more important than "Bait", so that it's parallel to the "Far" call for the puddles and a reminder to be closest to the boss.  If "Close Bait" is confusing or too long, I think dropping "Bait" would be better, since "Close" is more descriptive than just "Bait."

Reply:
> Hmm, I think for me, "Close" is actually more important than "Bait", so that it's parallel to the "Far" call for the puddles and a reminder to be closest to the boss. If "Close Bait" is confusing or too long, I think dropping "Bait" would be better, since "Close" is more descriptive than just "Bait."

I'm going to disagree completely here; to me, `Bait` is the more relevant part than `Close`.